packages/bun-usockets/src/udp.c

```diff
@@ -44,11 +42,15 @@ int us_udp_packet_buffer_payload_length(struct us_udp_packet_buffer_t *buf, int
 
 // what should we return? number of sent datagrams?
 int us_udp_socket_send(struct us_udp_socket_t *s, struct us_udp_packet_buffer_t *buf, int num) {
+    if (num == 0) return 0;
     int fd = us_poll_fd((struct us_poll_t *) s);
 
-    // we need to poll out if we failed
-
-    return bsd_sendmmsg(fd, buf, num, 0);
+    int sent = bsd_sendmmsg(fd, buf, num, 0);
+    if (0 <= sent && sent < num) {
+        // if we couldn't send all packets, register a writable event so we can call the drain callback
+        us_poll_change((struct us_poll_t *) s, s->loop, LIBUS_SOCKET_READABLE | LIBUS_SOCKET_WRITABLE);
```

Comment:
like with TCP sockets, we should just keep it polled imo

Reply:
keeping it polled results in a `drain` notification being delivered on every tick of the event loop, which isn't really helpful

Reply:
also, we don't expect to poll for writable often, because most uses will only send one packet anyway, which will always fit into the write buffer